rotkehlchen/accounting/export/csv.py

```diff
@@ -289,10 +292,28 @@ def create_zip(
 
         return success, filename
 
-    def to_csv_entry(self, event: 'ProcessedAccountingEvent') -> dict[str, Any]:
+    def to_csv_entry(self, event: Union['ProcessedAccountingEvent', 'HistoryBaseEntry']) -> dict[str, Any]:  # noqa: E501
         """Prepare the provided event to have a common format for the accounting
         CSV exported file.
         """
+        if isinstance(event, HistoryBaseEntry):
+            dict_event = {
+                'identifier': event.identifier,
+                'entry_type': event.entry_type.serialize(),
+                'event_identifier': event.event_identifier,
```

Comment:
This looks really wrong. You have one method that is for exporting the accounting and your are adding the check for the type here and then breaking the logic. You can move this to the `HistoryBaseEntry` class and add this as method. Also you are exporting the base fields but evm events and staking events have other fields that you also have to consider

Reply:
done